internal/provider/resources/block.go

```diff
@@ -116,16 +120,117 @@ func (r *BlockResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 			"workspace_id": schema.StringAttribute{
 				Optional:    true,
 				CustomType:  customtypes.UUIDType{},
-				Description: "Workspace ID (UUID) where the Block is located",
+				Description: "Workspace ID (UUID) where the Block is located. In Prefect Cloud, either the resource or the provider's `workspace_id` must be set in order to manage the Block.",
 			},
 		},
 	}
 }
 
+// copyBlockToModel maps an API response to a model that is saved in Terraform state.
+// A model can be a Terraform Plan, State, or Config object.
+func copyBlockToModel(block *api.BlockDocument, tfModel *BlockResourceModel) diag.Diagnostics {
+	// NOTE: we will map the `data` key OUTSIDE of this helper function, as we will
+	// need to skip this step for the POST /block_documents endpoint,
+	// which always returns masked data + will create inconsistent state between the
+	// plan <> fetched value - for Create(), we'll fall back to the user-configured JSON payload,
+	// whereas for Read() / Update() we can ask the API for unmasked values to ensure a consistent
+	// state drift check.
+	tfModel.ID = types.StringValue(block.ID.String())
+	tfModel.Created = customtypes.NewTimestampPointerValue(block.Created)
+	tfModel.Updated = customtypes.NewTimestampPointerValue(block.Updated)
+	tfModel.Name = types.StringValue(block.Name)
+	tfModel.TypeSlug = types.StringValue(block.BlockType.Slug)
+
+	return nil
+}
+
 // Create will create the Block resource through the API and insert it into the State.
-//
-//nolint:revive // TODO: remove this comment when method is implemented
 func (r *BlockResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var config BlockResourceModel
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	blockTypeClient, err := r.client.BlockTypes(config.AccountID.ValueUUID(), config.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Block Types", err))
+
+		return
```

Comment:
i got held up here bc i forgot to return after the client init errors.....kept running into weird nil pointer exceptions before i realized that i wasn't passing forward a non-nil client if a validation failed 🤦